CLN: Updated versions of `flake8` and `pre-commit-hooks` and modified files accordingly

Sai-Suraj-27 opened the pull request:
## Description of changes
 **Improvements & Bug fixes**
- Currently, most of the dependencies in `.pre-commit.config.yaml` file are not upto date. So, I updated them 
(`flake8` and `pre-commit-hooks`) to their latest versions. Also, the most of the pre-commit hooks are failing. So, i fixed the following:
	 1. `trailing-whitespace` hook by running `pre-commit run --all-files`.
	 2. `fix end of files` hook by running `pre-commit run --all-files`.
	 3. Fixed some flake8 errors like `E721 : Do not compare types use isinstance()`.
	 4. Added `F401 : module imported but unused` rule in extend-ignore because it's failing in a quite a few places and as it is not so important rule.
	
- New functionality
	 - ...

## Test plan
*How are these changes tested?*

- [ ] Tests pass locally with `pytest` for python, `yarn test` for js

## Documentation Changes
No need of any documentation changes.

beggers commented:
Thank you for doing this @Sai-Suraj-27!

I'd like us to keep `F401 : module imported but unused` warnings. You're right that it's not so important but over time it'll make our codebase less readable and, more importantly, possibly inflate Chroma's startup time.

Sai-Suraj-27 commented:
> Thank you for doing this @Sai-Suraj-27!
> 
> I'd like us to keep `F401 : module imported but unused` warnings. You're right that it's not so important but over time it'll make our codebase less readable and, more importantly, possibly inflate Chroma's startup time.

Hey, @beggers. Thanks for the review. So, should I remove the `F401 rule` from `extend-ignore` and update the PR by removing all the unused imports?

beggers commented:
Please remove the `F401 rule` from `extend-ignore`, yes! If you could also remove all unused imports in the codebase that would be so, so sick. But I would be happy to check this in as-is with that rule reenabled -- I don't want perfect to be the enemy of good :)

Sai-Suraj-27 commented:
Hey, @beggers. I removed the `F401 rule` from `extend-ignore` and removed all the `unused imports` except one (I didn't removed it because it was used in a commented out code in the file).

beggers reviewed (approved):
Thank you, this looks great! Really appreciate your work cleaning things up.
